src/installer/corehost/cli/apphost/static/singlefilehost.def

```diff
@@ -0,0 +1,6 @@
+; Licensed to the .NET Foundation under one or more agreements.
+; The .NET Foundation licenses this file to you under the MIT license.
+; See the LICENSE file in the project root for more information.
+
+EXPORTS
+    DllMain
```

Comment:
What is the point in exporting `DllMain` for .exe ? Does it even do anything?

Reply:
https://www.bing.com/search?q=Do%20I%20need%20to%20export%20DllMain

Reply:
A good part of coreclr initialization happens in DllMain, prior to calling coreclr_initialize. PAL guarantees that DllMain will be called, even for exe, as long as it can find exported DllMain.

Singlefilehost does not define its own DllMain. Here it exposes the one that is in coreclr, so it will still be called.

Reply:
JIT is different BTW. Since it always supported dll vs. embedded scenario there are different ways to initialize, controlled by `FEATURE_MERGE_JIT_AND_ENGINE`

Reply:
This file is used on Windows. Why do we need it on Windows?

For Unix, it would be nice to just call the initialization directly, without exporting DllMain from the app, etc.

Reply:
Actually, I am not sure if this works on Windows. It is required to export DllMain on Unix and Windows just has it the same way. 
On Windows we may still need to figure the mechanism for how this DllMain gets to run.

Reply:
>  I am not sure if this works on Windows

I am pretty sure that it does not.

> It is required to export DllMain on Unix

It is not required by things outside out control. It is our own self-imposed complexity.

Reply:
> It is not required by things outside out control. It is our own self-imposed complexity.

I agree on this. I kind of accepted this as the way we do things, but it feels very specific.
- what if we have other DllMain visible from the host executable? 
- guaranteeing that we have one and it is from coreclr seems somewhat a complicated contract?
- what solution will we have for this on Windows?

I would prefer if we had an extra API that host calls, prior to `coreclr_initialize` which causes coreclr to run DllMain or equivalent.
In fact - since we already have initializing entry point, is it possible that `coreclr_initialize` just run its DllMain if it has not run yet? 

I am going to try this.

Reply:
Are you planning to tackle this in a follow-up PR? You've marked it as "Resolved", but I don't see any changes w.r.t. DllMain stuff here.

Reply:
We will need a mechanism that works on Windows. It may be something similar to what PAL does or maybe not based on DllMain at all. 
After the discussion here I realized that we do not want to change anything for Unix. Not in this change anyways. It would be a big change and what we have works. 

I left the export for DllMain on windows is just to match Unix.

Reply:
> I left the export for DllMain on windows is just to match Unix.

Exported DllMain on Windows is non-sensical. It won't ever do anything useful. There is no point to match Unix on this. Revert this.

Reply:
Will revert.

Reply:
I've reverted DllMain export on windows.